examples/advanced/Tutorial3/reconstruction/FairTestDetectorRecoTask.cxx

```diff
@@ -97,7 +97,9 @@ void FairTestDetectorRecoTask::Exec(Option_t* /*opt*/)
         TVector3 dpos(1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12));
 
         FairTestDetectorHit* hit = new ((*fHitArray)[ipnt]) FairTestDetectorHit(-1, -1, pos, dpos);
-        hit->AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), FairRootManager::Instance()->GetBranchId("FairTestDetectorDigi"), ipnt));
+        if ( FairRootManager::Instance() != NULL ) {
```

Comment:
I just create a PR with the correct implementation for the FairRootManager (#575)

>  But even with the above change the code is not safe. The initialization in the ctor fgInstance = this; is dangerous. For example, if the caller of the ctor puts the object on a stack which goes out of scope before the program exits, each subsequent call to FairRootManager::Instance() will eventually point to random memory.
@dennisklein "Instance()" is static so this is not an issue.